elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,245 @@
+    static get observedAttributes() {
+      return ['opened'];
+    }
+
+    /**
+     * Returns the first child which is a menuitem.
+     */
+    _firstMenuItem() {
+      let child = this.firstElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns the last child which is a menuitem.
+     */
+    _lastMenuItem() {
+      let child = this.lastElementChild;
+      while (child) {
+        if (this._isMenuItem(child)) {
+          return child;
+        }
+        child = next.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Checks if a node is a 'menuitem', 'menuitemcheckbox' or 'menuitemradio'.
+     */
+    _isMenuItem(node) {
+      return MENUITEM_ARIA_ROLES.includes(node.getAttribute('role'));
+    }
+
+    /**
+     * Returns a menu item coming after the one passed as an argument, or null.
+     */
+    _nextMenuItem(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (this._isMenuItem(node)) {
+          return next;
+        }
+        next = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns a menu item coming before the one passed as an argument, or null.
+     */
+    _previousMenuItem(node) {
+      let prev = node.previousElementSibling;
+      while (prev) {
+        if (this._isMenuItem(node)) {
+          return prev;
+        }
+        prev = prev.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Handles keyboard interaction.
+     */
+    _onKeyDown(event) {
+      let el = event.target;
+      switch (event.keyCode) {
+        case KEYCODE.DOWN:
+          // If arrow down, move to next item. Wrap if necessary.
+          event.preventDefault();
+          let next = this._nextMenuItem(el) || this._firstMenuItem();
+          next.focus();
+          break;
+       case KEYCODE.UP:
+         // If arrow up, move to previous item. Wrap if necessary.
+         event.preventDefault();
+         let prev = this._previousMenuItem(el) || this._lastMenuItem();
+         prev.focus();
+         break;
+       case KEYCODE.ESCAPE:
+         // If escape, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       case KEYCODE.TAB:
+         // If tab, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       default:
+         break;
+      }
+      // If letter key, move to an item which starts with that letter.
+      if (event.keyCode >= KEYCODE.A && event.keyCode <= KEYCODE.Z) {
+        for (let child of this.children) {
+          if (child.innerText.trim()[0] === event.key) {
+            event.preventDefault();
+            child.focus();
+            break;
+          }
+        }
+      }
+    }
+
+    open() {
```

Comment:
Would you mind making this a private method with an underscore? I think in general we've been trying to avoid exposing public methods as much as possible, especially when they mimic the functionality of properties. The reasoning is that it's much easier to pass state into properties and their corresponding attributes. So the element can be managed in a more declarative fashion, and it's easier for top-down data flow libraries to talk to them.